File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/s3/src/main/kotlin/com/kotlin/s3/CreateBucket.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.s3

import aws.sdk.kotlin.services.s3.S3Client
import aws.sdk.kotlin.services.s3.model.CreateBucketRequest
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <bucketName> 

    Where:
        bucketName - the name of the Amazon S3 bucket to create. The Amazon S3 bucket name must be unique, or an error occurs.
    """

   if (args.size != 1) {
        println(usage)
        exitProcess(0)
   }
    val bucketName = args[0]
    createNewBucket(bucketName)
}

    suspend fun createNewBucket(bucketName: String) {

        val request = CreateBucketRequest {
             bucket = bucketName
         }

        S3Client { region = "us-east-1" }.use { s3 ->
            s3.createBucket(request)
            println("$bucketName is ready")
        }
 }

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/php/example_code/s3/CreateBucket.php
<?php
/*
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0
*/

require 'vendor/autoload.php';

use Aws\S3\S3Client;
use Aws\Exception\AwsException;

/* ////////////////////////////////////////////////////////////////////////////
 * Purpose: Creates a bucket in Amazon S3.
 * 
 * Inputs:
 * - $s3Client: An initialized AWS SDK for PHP API client for S3.
 * - $bucketName: The name of the bucket to create.
 * 
 * Returns: Information about the bucket; otherwise, the error message.
 * ///////////////////////////////////////////////////////////////////////// */

function createBucket($s3Client, $bucketName)
{
    try {
        $result = $s3Client->createBucket([
            'Bucket' => $bucketName,
        ]);
        return 'The bucket\'s location is: ' .
            $result['Location'] . '. ' .
            'The bucket\'s effective URI is: ' . 
            $result['@metadata']['effectiveUri'];
    } catch (AwsException $e) {
        return 'Error: ' . $e->getAwsErrorMessage();
    }
}

function createTheBucket()
{
    $s3Client = new S3Client([
        'profile' => 'default',
        'region' => 'us-east-1',
        'version' => '2006-03-01'
    ]);

    echo createBucket($s3Client, 'my-bucket');
}

// Uncomment the following line to run this code in an AWS account.
// createTheBucket();

